src/fetchtastic/download/client_app.py

```
@@ -0,0 +1,1039 @@
+            return os.path.commonpath([download_root, candidate]) == download_root
+        except ValueError:
+            return False
+
+    def _is_safe_managed_dir(self, path: str) -> bool:
+        return (
+            os.path.isdir(path)
+            and not os.path.islink(path)
+            and self._is_within_download_tree(path)
+        )
+
+    def _move_legacy_path(self, source_path: str, destination_path: str) -> bool:
```

Ruff (0.15.12)

[warning] Too many return statements (7 > 6)

(PLR0911)